### PR TITLE
Added user posts page and applying new firebase design

### PR DESCRIPTION
**Pull request by arunstop**
- cleanup : clearing some debugging strings
- add : modal dialog for yes and no modal
- update : logging out now uses confirmation dialog
- refactor : renaming all modal related components
- refactor : renaming input related components
- fix : choose topic page redirections now are correct
- refactor : simplifying user registration api modules
- attempt : optimizing status placeholder application
- clean up : cleaning console
- clean up : minor console clean up
- add : date-fns package
- add : user/posts page and getting article by ids
- add : modal routed hook
- added : user/posts page also added article deletion
- update : made the modal confirmation enter/space-able
- update : minor ui fix
- update : big update on firebase, applying new system design on how article is stored to the firebase
- update : implemented the thing new firebase article design
- update : applying article deletion with the new firebase design
- fix : writing panel preview not showing the thumbnail
- update : article page now works with the implementation of new firebase config
- fix : article page now shows the proper article with newly firebase config applied
- fix : fix redirection after creating article, also add to link to /user/posts
- update : added search functionality  on user/posts

### CODE REVIEW

**Comment by scuffed-developer** [comment]
TODO : next

- add intersection observer to load article content
- add more detailed properties to post item mini
-

**Comment by arunstop** [comment]
I don't think intersection observer on article page would be a great implementation. Because of comments and suggestion is already a lazy loaded component. If the users scroll down they might as well accidentally load the comments and suggestions.

Though i guess i can make the comments and suggestion only available if the content is already loaded.

File: pages/user/posts.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
+import { Transition } from "@headlessui/react";
```
Comment (scuffed-developer) [inline pages/user/posts.tsx:1]: Probably need to code-split this, this is doing too much.

Comment (scuffed-developer) [reply pages/user/posts.tsx:1]: But i will pass it for now, moving on to the next one

**Review by scuffed-developer — APPROVED** [review]
All working as intended, i am going to approve this one, but you need to check it all again to look for optimization chance.

File: pages/user/posts.tsx (hunk 1 above)
Comment (arunstop) [reply pages/user/posts.tsx:1]: Okay!

**Pull request merged**